api/services/execution/execution.proto

```diff
@@ -106,3 +109,35 @@ message PauseRequest {
 message ResumeRequest {
 	string id = 1 [(gogoproto.customname) = "ID"];
 }
+
+message CheckpointRequest {
+	string id = 1 [(gogoproto.customname) = "ID"];
+	bool allow_tcp = 2;
+	bool allow_unix_sockets = 3;
+	bool allow_terminal = 4;
+	bool file_locks = 5;
+	repeated string empty_namespaces = 6;
+	string parent_checkpoint = 7 [(gogoproto.customtype) = "github.com/opencontainers/go-digest.Digest", (gogoproto.nullable) = false];
+	bool exit = 8;
+}
+
+message CheckpointResponse {
+	repeated types.Descriptor descriptors = 1;
+}
+
+message RestoreRequest {
```

Comment:
Should we overload the `Create` method? Looks like the input and results are the same except for the digest. It probably make sense to have these separate...